File: frontend/src/services/aiService.js
// AI Service — mock sentiment analysis
const delay = ms => new Promise(r => setTimeout(r, ms));

const POSITIVE_WORDS = ['great', 'excellent', 'amazing', 'love', 'fantastic', 'perfect', 'best', 'wonderful', 'outstanding', 'recommend'];
const NEGATIVE_WORDS = ['bad', 'terrible', 'worst', 'hate', 'awful', 'horrible', 'disappointed', 'useless', 'poor', 'pathetic'];

export async function analyzeSentiment(text) {
  await delay(1200); // simulate AI processing

  const lower = text.toLowerCase();
  let positiveCount = POSITIVE_WORDS.filter(w => lower.includes(w)).length;
  let negativeCount = NEGATIVE_WORDS.filter(w => lower.includes(w)).length;

  let sentiment, score, label;

  if (positiveCount > negativeCount + 1) {
    sentiment = 'positive';
    score = Math.min(95, 60 + positiveCount * 8 + Math.floor(Math.random() * 15));
    label = score > 85 ? 'Highly Positive & Authentic' : 'Positive with Valid Concerns';
  } else if (negativeCount > positiveCount + 1) {
    sentiment = 'negative';
    score = Math.max(15, 55 - negativeCount * 7 + Math.floor(Math.random() * 10));
    label = 'Negative — Specific Grievances';
  } else {
    sentiment = 'mixed';
    score = 40 + Math.floor(Math.random() * 20);
    label = 'Mixed — Credible Issues Identified';
  }

  return {
    sentiment,
    score,
    label,
    confidence: 0.87 + Math.random() * 0.10,
    isSpam: false,
    isFake: false,
    keywords: extractKeywords(text),
    engine: 'NVIDIA NIM (NeMo Microservice)',
    nvidiaAccelerated: true,
  };
}

function extractKeywords(text) {
  const words = text.toLowerCase().split(/\W+/).filter(w => w.length > 4);
  const freq = {};
  words.forEach(w => { freq[w] = (freq[w] || 0) + 1; });
  return Object.entries(freq)
    .sort((a, b) => b[1] - a[1])
    .slice(0, 5)
    .map(([word]) => word);
}

export async function detectFakeReview(text) {
  await delay(800);
  const wordCount = text.trim().split(/\s+/).length;
  const isFake = wordCount < 10 || /buy now|click here|promo code/i.test(text);
  return { isFake, confidence: isFake ? 0.91 : 0.08 };
}

File: backend/src/main/java/com/rivyuu/connect/service/AIService.java
package com.rivyuu.connect.service;

import org.springframework.stereotype.Service;

import java.util.Arrays;
import java.util.List;

/**
 * AI Service — Sentiment analysis and fake review detection.
 * In production, this would call a Python NLP microservice (e.g., HuggingFace Transformers).
 * For demo, uses keyword-based heuristics.
 */
@Service
public class AIService {

    private static final List<String> POSITIVE_WORDS = Arrays.asList(
            "great", "excellent", "amazing", "love", "fantastic", "perfect", "best",
            "wonderful", "outstanding", "recommend", "superb", "brilliant", "awesome"
    );

    private static final List<String> NEGATIVE_WORDS = Arrays.asList(
            "bad", "terrible", "worst", "hate", "awful", "horrible", "disappointed",
            "useless", "poor", "pathetic", "scam", "fraud", "broken", "waste"
    );

    private static final List<String> SPAM_PATTERNS = Arrays.asList(
            "buy now", "click here", "promo code", "free money", "100% guaranteed",
            "limited offer", "act now", "special deal"
    );

    /**
     * Analyze sentiment: returns "positive", "negative", or "mixed"
     */
    public String analyzeSentiment(String text) {
        String lower = text.toLowerCase();
        long positiveCount = POSITIVE_WORDS.stream().filter(lower::contains).count();
        long negativeCount = NEGATIVE_WORDS.stream().filter(lower::contains).count();

        if (positiveCount > negativeCount + 1) return "positive";
        if (negativeCount > positiveCount + 1) return "negative";
        return "mixed";
    }

    /**
     * Compute a 0-100 trust/sentiment score
     */
    public Integer computeScore(String text, String sentiment) {
        String lower = text.toLowerCase();
        long positiveCount = POSITIVE_WORDS.stream().filter(lower::contains).count();
        long negativeCount = NEGATIVE_WORDS.stream().filter(lower::contains).count();
        int wordCount = text.trim().split("\\s+").length;

        int base;
        switch (sentiment) {
            case "positive" -> base = 65 + (int) Math.min(30, positiveCount * 8);
            case "negative" -> base = 40 - (int) Math.min(25, negativeCount * 7);
            default -> base = 45;
        }

        // Bonus for detailed reviews
        if (wordCount > 50) base += 5;
        if (wordCount > 100) base += 5;

        return Math.max(5, Math.min(100, base));
    }

    /**
     * Detect if a review is spam/fake
     */
    public boolean isFakeReview(String text) {
        String lower = text.toLowerCase();
        int wordCount = text.trim().split("\\s+").length;
        boolean hasSpamPattern = SPAM_PATTERNS.stream().anyMatch(lower::contains);
        return wordCount < 10 || hasSpamPattern;
    }

    /**
     * Generate AI label from sentiment and score
     */
    public String generateLabel(String sentiment, int score) {
        return switch (sentiment) {
            case "positive" -> score >= 85 ? "Highly Positive & Authentic" : "Positive with Valid Concerns";
            case "negative" -> "Negative — Specific Grievances";
            default -> "Mixed — Credible Issues Identified";
        };
    }
}
